src/main/java/sg/dex/crypto/Hash.java

```diff
@@ -4,7 +4,11 @@
 import org.bouncycastle.jcajce.provider.digest.SHA3;
 import sg.dex.starfish.util.Hex;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
```

Comment:
Don't use BigInteger unless absolutely necessary. Slow and error prone due to sign issues etc.